### PR TITLE
Add WAN support for ICache.putAll

### PR DESCRIPTION
**Pull request by mmedenjak**
Added support for publishing WAN events in the case of ICache.putAll.
This requires changing the return type on a couple of methods of the
record store but the return type was never used so this probably
isn't an issue.

Fixes : https://github.com/hazelcast/hazelcast-enterprise/issues/1531
EE : https://github.com/hazelcast/hazelcast-enterprise/pull/1539

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllBackupOperation.java [hunk 1]
```diff
@@ -75,10 +78,26 @@ public void run() throws Exception {
...
+            ICacheService service = getService();
+            final CacheWanEventPublisher publisher = service.getCacheWanEventPublisher();
+            final CacheEntryView<Data, Data> view = CacheEntryViews.createDefaultEntryView(
+                    key, toData(record.getValue()), record);
+            publisher.publishWanReplicationUpdate(name, view);
+        }
+    }
+
+    private Data toData(Object o) {
+        return getNodeEngine().getSerializationService().toData(o);
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllBackupOperation.java]: was the intent to create heap-data from native one? if yes, this shouldn't work, instead `ToHeapDataConverter` can be used.

File: hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllOperation.java [hunk 2]
```diff
@@ -86,6 +88,16 @@ public void run()
             Data value = entry.getValue();
             CacheRecord backupRecord = cache.put(key, value, expiryPolicy, callerUuid, completionId);
             backupRecords.put(key, backupRecord);
+
+            publishWanEvent(key, value, backupRecord);
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllOperation.java]: same here if to-heap-data conversion is needed

File: hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllBackupOperation.java (hunk 1 above)
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllBackupOperation.java]: ok i see this works

**Comment by jerrinot** [comment]
run-lab-run

**Pull request merged**